### PR TITLE
[feat]: [IAC-513]: Fix NPE and support the use of image/use tags

### PR DESCRIPTION
**Pull request by nassergonzalez**
## Describe your changes
The original ticket was to fix a NPE when trying to use the `uses` yaml tag. After fixing it, we realised that we were missing the proper use of the `uses` and `image` tag so this PR also includes that.
On top of that, it contains a change in the IACMManager that is needed to be included in order to not have the Pipeline been stuck in the Initialise step

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/47049)

### CODE REVIEW

**Comment by nassergonzalez** [comment]
trigger messagecheck

**Comment by nassergonzalez** [comment]
trigger messagecheck

File: 332-ci-manager/service/src/main/java/io/harness/iacm/execution/IACMStepsUtils.java [hunk 1]
```diff
@@ -123,68 +117,62 @@ private StackVariables[] getIACMStackVariables(String org, String projectId, Str
...
             CIStepInfoType.IACM_TERRAFORM, ngAccess.getAccountIdentifier());
       default:
         return null;
     }
   }
-  public VmPluginStep injectIACMInfo(Ambiance ambiance, PluginStepInfo stepInfo, StageInfraDetails stageInfraDetails,
-      ParameterField<Timeout> parameterFieldTimeout) {
-    long timeout = TimeoutUtils.getTimeoutInSeconds(parameterFieldTimeout, stepInfo.getDefaultTimeout());
 
+  public Object[] AddIACMInfoToStep(Ambiance ambiance, PluginStepInfo stepInfo) {
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/iacm/execution/IACMStepsUtils.java:147]: AddIACMInfoToStep --> addIACMInfoToStep

Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/iacm/execution/IACMStepsUtils.java:147]: split it into 2 methods to envvars and connnector ref

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/serializer/vm/VmPluginStepSerializer.java [hunk 2]
```diff
@@ -124,8 +124,10 @@ public VmStepInfo serialize(PluginStepInfo pluginStepInfo, StageInfraDetails sta
     }
     envVars.putAll(resolveMapParameterV2("envVars", "pluginStep", identifier, pluginStepInfo.getEnvVariables(), false));
 
-    String image =
-        RunTimeInputHandler.resolveStringParameter("Image", stepName, identifier, pluginStepInfo.getImage(), false);
+    if (Objects.equals(image, "")) {
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/serializer/vm/VmPluginStepSerializer.java:127]: use isempty

**Review by harshjain12 — DISMISSED** [review]

**Comment by nassergonzalez** [comment]
trigger utAll

**Review by harshjain12 — APPROVED** [review]

**Review by scottyw — COMMENTED** [review]
> Reviewed 1 of 5 files at r1, 2 of 2 files at r2, 3 of 4 files at r3, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47049)* status: :shipit: complete! ✓

**Pull request merged**